datalad/__init__.py

```diff
@@ -12,6 +12,14 @@
 
 # Other imports are interspersed with lgr.debug to ease troubleshooting startup
 # delays etc.
+
+# If there is a bundled git, make sure GitPython uses it too:
+from datalad.cmd import GitRunner
+_git_runner = GitRunner()
+if _git_runner._GIT_PATH:
+    import os
+    os.environ['GIT_PYTHON_GIT_EXECUTABLE'] = os.path.join(_git_runner._GIT_PATH, 'git')
```

Comment:
Edited above, since I was arguing wrt to this being part of `GitRepo`'s constructor (which would be even worse). But that's probably not what you meant anyway. ;-)